### PR TITLE
Added Centrality Algorithms Example Jupyter Notebook

### PR DESCRIPTION
**Pull request by kedarghule**
This PR adds an example jupyter notebook for centrality algorithms on the airline travel reachability network dataset. The notebook explores eigenvector centrality, betweenness centrality & degree centrality.

This is in reference to my comment in the issue: https://github.com/neo4j/graph-data-science-client/issues/254

Thank you :)

### CODE REVIEW

**Comment by kedarghule** [comment]
Umm. I don't know why one check is not running :/

**Comment by Mats-SX** [comment]
@kedarghule The check not running is normal. For security reasons we do not run PR builds from non-members. There have been several attacks made on our public repositories to attempt to expose secrets from our build system.

I apologise for the delayed review. We are in the process of releasing GDS `2.4.0` which is taking up most of my time lately. We are grateful for your contribution, and you are not forgotten :)

All the best
Mats

**Comment by kedarghule** [comment]
> @kedarghule The check not running is normal. For security reasons we do not run PR builds from non-members. There have been several attacks made on our public repositories to attempt to expose secrets from our build system.
> 
> I apologise for the delayed review. We are in the process of releasing GDS `2.4.0` which is taking up most of my time lately. We are grateful for your contribution, and you are not forgotten :)
> 
> All the best Mats

Alright @Mats-SX  and sure, no problem :)

**Review by Mats-SX — APPROVED** [review]
Thanks a lot for this contribution! It's a great notebook. I only have a few comments.

I will push a commit that clears output cells and applies our code style rules to the Python code. For reference, we have two scripts in the `scripts` directory: `checkstyle` and `makestyle` that can be used (just run them as `scripts/checkstyle` without arguments) to manage this in future contributions.

I will also create a sibling PR to allow our CI to run.

File: examples/centrality-algorithms.ipynb [hunk 1]
```diff
@@ -0,0 +1,991 @@
...
+   "source": [
+    "routes_df = pd.read_csv('https://snap.stanford.edu/data/reachability.txt.gz', sep=' ', skiprows=6, header=None, compression='gzip', names=['Origin', 'Destination', 'Weight'])\n",
+    "routes_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we load this data (nodes and edges) into a Graph Database and a GDS graph."
```
[the remark this answers is not in the document]
Comment (kedarghule) [reply examples/centrality-algorithms.ipynb]: Alright. Done :)

File: examples/centrality-algorithms.ipynb [hunk 2]
```diff
@@ -0,0 +1,991 @@
...
+    }
+   ],
+   "source": [
+    "G, result = gds.graph.project(\"airline\", \"City\", \"HAS_FLIGHT_TO\")\n",
+    "\n",
+    "print(f\"The projection took {result['projectMillis']} ms\")\n",
+    "\n",
+    "# We can use convenience methods on `G` to check if the projection looks correct\n",
+    "print(f\"Graph '{G.name()}' node count: {G.node_count()}\")\n",
+    "print(f\"Graph '{G.name()}' node labels: {G.node_labels()}\")"
```
[the remark this answers is not in the document]
Comment (kedarghule) [reply examples/centrality-algorithms.ipynb]: Done!

File: examples/centrality-algorithms.ipynb [hunk 3]
```diff
@@ -0,0 +1,991 @@
...
+   "source": [
+    "### References\n",
+    "- For the network:\n",
+    "Brendan J. Frey and Delbert Dueck. \"Clustering by passing messages between data points.\" Science 315.5814 (2007): 972-976.\n",
+    "\n",
+    "- For the city metadata (metropolitan population, latitude, and longitude):\n",
+    "Austin R. Benson, David F. Gleich, and Jure Leskovec. \"Higher-order Organization of Complex Networks.\" Science, 353.6295 (2016): 163–166.\n",
+    "\n",
+    "- Notebook contributed by [Kedar Ghule](https://github.com/kedarghule)"
+   ]
```
[the remark this answers is not in the document]
Comment (kedarghule) [reply examples/centrality-algorithms.ipynb]: Done!

**Comment by kedarghule** [comment]
@Mats-SX I have addressed your comments for the PR and made a new commit. Please let me know if anything else is needed or if I missed anything.

Also the latest commit won't run a check again :/

File: examples/centrality-algorithms.ipynb [hunk 4]
```diff
@@ -0,0 +1,991 @@
...
+       " 'min': 0.0012630745768547058,\n",
+       " 'max': 0.0856785699725151,\n",
+       " 'mean': 0.041094380038741385,\n",
+       " 'p90': 0.07575749605894089,\n",
+       " 'p50': 0.0386042520403862,\n",
+       " 'p999': 0.0856785699725151,\n",
+       " 'p95': 0.08167456835508347,\n",
+       " 'p75': 0.05822562426328659}"
+      ]
+     },
```
[the remark this answers is not in the document]
Comment (Mats-SX) [reply examples/centrality-algorithms.ipynb]: Yeah I noticed that our `makestyle` script does this automatically :)

File: examples/centrality-algorithms.ipynb [hunk 5]
```diff
@@ -438,13 +462,29 @@
...
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.10.11"
```
Comment (Mats-SX) [inline examples/centrality-algorithms.ipynb]: This stuff was added back in, which is normal when the notebook is run. I will clear it out again.

**Comment by Mats-SX** [comment]
@kedarghule Thank you! I've kicked off another CI build.

**Pull request merged**